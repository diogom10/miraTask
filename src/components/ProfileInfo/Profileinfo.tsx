import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';
import {Warning} from '../../assets/icons/warning';
import {Profile} from '../../models/Profile';
import {PhoneIcon} from "../../assets/icons/phone";
import {EmailIcon} from "../../assets/icons/email";

export interface IProfileInfoProps {
  profile: Profile;
  currentTasks: number;
  completedTasks: number;
  testID?: string;
}

export const ProfileInfo = ({
  profile,
  currentTasks = 0,
  completedTasks = 0,
  testID = 'ProfileInfo',
}: IProfileInfoProps) => {
  return (
    <>
      <View style={styles.containerHeader}>
        <View style={styles.userInfo}>
          <Image
            testID={`${testID}_profile_image`}
            style={styles.image}
            source={{
              uri:
                profile?.phoroUrl ||
                'https://www.nicepng.com/png/detail/933-9332131_profile-picture-default-png.png',
            }}
          />
          <View style={styles.controlName}>
            <Text style={styles.title} testID={`${testID}_name`}>
              {profile?.name}
            </Text>
            <Text style={styles.subtitle} testID={`${testID}_profession`}>
              {profile?.profession}
            </Text>
          </View>
        </View>
        <View style={styles.contactInfo}>
          <View style={styles.rowInfo}>
            <PhoneIcon width={16} height={16} fill={'#989aae'} />
            <Text style={styles.infoText} testID={`${testID}_phone`}>
              {profile?.phone}
            </Text>
          </View>
          <View style={styles.rowInfo}>
            <EmailIcon width={16} height={16} fill={'#989aae'} />
            <Text style={styles.infoText} testID={`${testID}_email`}>
              {profile?.email}
            </Text>
          </View>
        </View>
      </View>
      <View style={styles.containerTaskInfo}>
        <View style={styles.taskInfo}>
          <Text style={styles.title}>Current Tasks</Text>
          <Text style={styles.subtitle} testID={`${testID}_current_tasks`}>
            {currentTasks}
          </Text>
        </View>
        <View style={styles.taskInfoSeparatorCenter} />
        <View style={styles.taskInfo}>
          <Text style={styles.title}>Completed Tasks</Text>
          <Text style={styles.subtitle} testID={`${testID}_completed_tasks`}>{completedTasks}</Text>
        </View>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  containerHeader: {
    padding: 20,
    borderRadius: 20,
    margin: 2,
    minHeight: 70,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  controlName: {
    flexDirection: 'column',
    marginLeft: 40,
  },
  image: {
    width: 80,
    height: 80,
    borderRadius: 50,
    overflow: 'hidden',
  },
  title: {
    fontSize: 16,
    color: '#0b1352',
    fontWeight: 'bold',
  },
  subtitle: {
    color: '#989aae',
  },
  contactInfo: {
    marginTop: 20,
  },
  rowInfo: {
    flexDirection: 'row',
    marginBottom: 10,
    alignItems: 'center',
  },
  infoText: {
    marginLeft: 10,
    color: '#989aae',
  },
  containerTaskInfo: {
    minHeight: 90,
    width: '100%',
    justifyContent: 'space-between',
    borderColor: '#ebecee',
    borderTopWidth: 3,
    borderBottomWidth: 3,
    borderRightWidth: 0,
    borderLeftWidthWidth: 0,
    flexDirection: 'row',
  },
  taskInfo: {
    width: '49%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  taskInfoSeparatorCenter: {
    height: '100%',
    borderWidth: 2,
    borderColor: '#ebecee',
  },
});
