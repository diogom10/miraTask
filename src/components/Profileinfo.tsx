import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';
import {Warning} from '../assets/icons/warning';
import {Profile} from '../models/Profile';

interface IProfileInfoProps {
  profile: Profile;
  currentTasks: number;
  completedTasks: number;
}

export const ProfileInfo = ({
  profile,
  currentTasks = 0,
  completedTasks = 0,
}: IProfileInfoProps) => {
  return (
    <>
      <View style={styles.containerHeader}>
        <View style={styles.userInfo}>
          <Image
            style={styles.image}
            source={{
              uri:
                profile?.phoroUrl ||
                'https://www.nicepng.com/png/detail/933-9332131_profile-picture-default-png.png',
            }}
          />
          <View style={styles.controlName}>
            {/*<Text style={styles.title}>Diogo Moura</Text>*/}
            {/*<Text style={styles.subtitle}>React Native Developer</Text>  */}
            <Text style={styles.title}>{profile?.name}</Text>
            <Text style={styles.subtitle}>{profile?.profession}</Text>
          </View>
        </View>
        <View style={styles.contactInfo}>
          <View style={styles.rowInfo}>
            <Warning width={16} height={16} fill={'#989aae'} />
            <Text style={styles.infoText}>{profile?.phone}</Text>
          </View>
          <View style={styles.rowInfo}>
            <Warning width={16} height={16} fill={'#989aae'} />
            <Text style={styles.infoText}>{profile?.email}</Text>
          </View>
        </View>
      </View>
      <View style={styles.containerTaskInfo}>
        <View style={styles.taskInfo}>
          <Text style={styles.title}>Current Tasks</Text>
          <Text style={styles.subtitle}>{currentTasks}</Text>
        </View>
        <View style={styles.taskInfoSeparatorCenter} />
        <View style={styles.taskInfo}>
          <Text style={styles.title}>Completed Tasks</Text>
          <Text style={styles.subtitle}>{completedTasks}</Text>
        </View>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  containerHeader: {
    padding: 20,
    borderRadius: 20,
    margin: 2,
    minHeight: 70,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  controlName: {
    flexDirection: 'column',
    marginLeft: 40,
  },
  image: {
    width: 80,
    height: 80,
    borderRadius: 50,
    overflow: 'hidden',
  },
  title: {
    fontSize: 16,
    color: '#0b1352',
    fontWeight: 'bold',
  },
  subtitle: {
    color: '#989aae',
  },
  contactInfo: {
    marginTop: 20,
  },
  rowInfo: {
    flexDirection: 'row',
    marginBottom: 10,
    alignItems: 'center',
  },
  infoText: {
    marginLeft: 10,
    color: '#989aae',
  },
  containerTaskInfo: {
    minHeight: 90,
    width: '100%',
    justifyContent: 'space-between',
    borderColor: '#ebecee',
    borderTopWidth: 3,
    borderBottomWidth: 3,
    borderRightWidth: 0,
    borderLeftWidthWidth: 0,
    flexDirection: 'row',
  },
  taskInfo: {
    width: '49%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  taskInfoSeparatorCenter: {
    height: '100%',
    borderWidth: 2,
    borderColor: '#ebecee',
  },
});
