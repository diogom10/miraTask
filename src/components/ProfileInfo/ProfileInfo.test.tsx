import React from 'react';
import {create} from 'react-test-renderer';
import {IProfileInfoProps, ProfileInfo} from './Profileinfo';

const props: IProfileInfoProps = {
  completedTasks: 3,
  currentTasks: 10,
  profile: {
    email: 'diogomoura10@gmail.com',
    name: 'Diogo Moura',
    phone: '(+55) 11 983593722',
    phoroUrl: 'https://randomuser.me/api/portraits/men/42.jpg',
    profession: 'React Native Developer',
  },
  testID: 'ProfileInfo',
};

const component = create(<ProfileInfo {...props} />);
const el_profile_image = `${props.testID}_profile_image`;
const el_profession = `${props.testID}_profession`;
const el_name = `${props.testID}_name`;
const el_email = `${props.testID}_email`;
const el_current_tasks = `${props.testID}_current_tasks`;
const el_completed_tasks = `${props.testID}_completed_tasks`;

describe('TaskView Component', () => {
  test(`should render image #${el_profile_image}`, async () => {
    const image = component.root.findByProps({testID: el_profile_image}).props;
    expect(image).toBeTruthy();
  });
  test(`should render label #${el_profession}`, async () => {
    const label = component.root.findByProps({testID: el_profession}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_name}`, async () => {
    const label = component.root.findByProps({testID: el_name}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_email}`, async () => {
    const label = component.root.findByProps({testID: el_email}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_current_tasks}`, async () => {
    const label = component.root.findByProps({testID: el_current_tasks}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_completed_tasks}`, async () => {
    const label = component.root.findByProps({
      testID: el_completed_tasks,
    }).props;
    expect(label).toBeTruthy();
  });
  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
