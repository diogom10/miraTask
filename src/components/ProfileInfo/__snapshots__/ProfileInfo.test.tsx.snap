// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView Component snapshot 1`] = `
[
  <View
    style={
      {
        "borderRadius": 20,
        "margin": 2,
        "minHeight": 70,
        "padding": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://randomuser.me/api/portraits/men/42.jpg",
          }
        }
        style={
          {
            "borderRadius": 50,
            "height": 80,
            "overflow": "hidden",
            "width": 80,
          }
        }
        testID="ProfileInfo_profile_image"
      />
      <View
        style={
          {
            "flexDirection": "column",
            "marginLeft": 40,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0b1352",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
          testID="ProfileInfo_name"
        >
          Diogo Moura
        </Text>
        <Text
          style={
            {
              "color": "#989aae",
            }
          }
          testID="ProfileInfo_profession"
        >
          React Native Developer
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="#989aae"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={16}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4288191150,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-1 6h2v8h-2v-8zm1 12.25c-.69 0-1.25-.56-1.25-1.25s.56-1.25 1.25-1.25 1.25.56 1.25 1.25-.56 1.25-1.25 1.25z"
              fill={
                {
                  "payload": 4288191150,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            {
              "color": "#989aae",
              "marginLeft": 10,
            }
          }
          testID="ProfileInfo_phone"
        >
          (+55) 11 983593722
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          fill="#989aae"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={16}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4288191150,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-1 6h2v8h-2v-8zm1 12.25c-.69 0-1.25-.56-1.25-1.25s.56-1.25 1.25-1.25 1.25.56 1.25 1.25-.56 1.25-1.25 1.25z"
              fill={
                {
                  "payload": 4288191150,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            {
              "color": "#989aae",
              "marginLeft": 10,
            }
          }
          testID="ProfileInfo_email"
        >
          diogomoura10@gmail.com
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "borderBottomWidth": 3,
        "borderColor": "#ebecee",
        "borderLeftWidthWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 3,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 90,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "49%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#0b1352",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Current Tasks
      </Text>
      <Text
        style={
          {
            "color": "#989aae",
          }
        }
        testID="ProfileInfo_current_tasks"
      >
        10
      </Text>
    </View>
    <View
      style={
        {
          "borderColor": "#ebecee",
          "borderWidth": 2,
          "height": "100%",
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "49%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#0b1352",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Completed Tasks
      </Text>
      <Text
        style={
          {
            "color": "#989aae",
          }
        }
        testID="ProfileInfo_completed_tasks"
      >
        3
      </Text>
    </View>
  </View>,
]
`;
